segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/TLSConfigChangeEventConsumer.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host.handler;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@RequiredArgsConstructor
+@Slf4j
+public class TLSConfigChangeEventConsumer implements Consumer<WatchEvent<?>> {
+
+    /**
+     * A counter representing the number of times this object has been asked to
+     * consume an event.
+     */
+    private static final AtomicInteger NUM_OF_CONFIG_CHANGES_SINCE_START = new AtomicInteger(0);
+
+    private @NonNull final Channels channels;
+
+    @Override
+    public void accept(WatchEvent<?> watchEvent) {
```

Comment:
Where do you use this argument?

Reply:
The `FileModificationWatcher` object to which we pass an object of `TLSConfigChangeEventConsumer` as a callback, provides the `WatchEvent` as context to the callback. Currently `TLSConfigChangeEventConsumer` uses the `WatchEvent` only for logging the name of the file for which the modification was detected. 

The newly introduced `FileModificationWatcher` is a generic class that can be useful in other places too, and I wanted to keep its interface useful for general use. So, I'm passing the WatchEvent as a callback. 

What do you say?